### ISSUE TITLE
Doesn't build on non-GNU systems due to use of parallel/numeric

### ISSUE THREAD

**Issue by yurivict** [KarlsruheMIS/KaMIS#27]
```parallel/numeric``` is Linux-only.

**Comment by Hespian** [comment]
parallel/numeric seems to only be used in the maximum matching code ([lib/mis/kernel/ParFastKer/fast_reductions/src/MaximumMatching.cpp](https://github.com/KarlsruheMIS/KaMIS/blob/f99641d719594b4f7091898c9002a8dc762213d2/lib/mis/kernel/ParFastKer/fast_reductions/src/MaximumMatching.cpp)). In `parallel/*` there are some stl algorithms implemented in parallel - in particular, here only `partial_sum` is used. If you just want to make this compile on non-GNU systems, you could just replace `<parallel/numeric>` with `<numeric>` and `__gnu_parallel::partial_sum` with `std:partial_sum`. But be aware that this will probably slow down kernelization a bit because it would then compute prefix sums sequentially.

**Comment by yurivict** [comment]
I am trying to create the FreeBSD port for MaMIS. There's no ```parallel/numeric``` header.
Is there a way to build without it?

**Comment by yurivict** [comment]
This advise seems to work. Thanks!

**Comment by darrenstrash** [comment]
Hi @yurivict, I've written and am testing an algorithm that will make parallel/numeric no longer needed. I would propose that we discuss making the FreeBSD port. We would like to make sure that the port is as efficient as our original implementation (i.e., to ensure the running times are consistent with our papers). Can I help with this endeavor? If so, is it possible to commit all of the necessary changes to the github repo, or is there much more that needs to be done for the port?

**Comment by darrenstrash** [comment]
Thanks @yurivict

Just to check: In the commit info, are those the only changes that you had to make from our git repo to your port?

Re testsuite: While some of the algorithms are deterministic, several of them are actually non-deterministic and therefore hard to test. The main performance metric we care about, time, is difficult to write automated tests for. I will try out your port and let you know if I see performance similar to our papers.

**Comment by schulzchristian** [comment]
@darrenstrash can we close this issue?

**Comment by darrenstrash** [comment]
Let's not close it yet. I've self assigned it and added it to my task list for my winter break. If I don't complete it by Jan 31, I'll close it.
